ruby/vampires.rb

```diff
@@ -1,30 +1,63 @@
-puts "Enter First Name"
-firstName = gets.chomp
-puts "Enter Middle Initial"
-middleInitial = gets.chomp
-puts "Enter Last Name"
-lastName = gets.chomp
-puts "Enter Age"
-applicantAge = gets.chomp
-puts "Enter Year Born"
-birthYear = gets.chomp
-puts "Would you like some Garlic Bread fresh from our cafeteria?"
-garlicReply = gets.chomp
-puts "Would you like to enroll in the health insurance plan offered by the company?"
-insuranceReply = gets.chomp
-currentYear = Time.new.year
-birthYear - currentYear = ageCheck
-firstName + lastName = fullName
+puts "Welcome to the Werewolf Resources Department"
+puts "How many applicants are being processed at this time?"
+numApplicants = gets.chomp.to_i
+for i in 0..(numApplicants-1)
+  puts "Enter First Name"
+  firstName = gets.chomp
+  puts "Enter Last Name"
+  lastName = gets.chomp
+  puts "Enter Age"
+  applicantAge = gets.chomp
+  puts "Enter Year Born"
+  birthYear = gets.chomp.to_i
+  puts "Would you like some garlic bread fresh from our cafeteria?  Answer y/n"
+  garlicReply = gets.chomp.downcase
+  puts "Would you like to enroll in the health insurance plan? Answer y/n "
+  insuranceReply = gets.chomp.downcase
 
-if ageCheck == applicantAge && (garlicReply == true || insuranceReply == true)
-  puts "Probably not a vampire"
-elsif ageCheck != applicantAge && (garlicReply == false || insuranceReply == false)
-  puts "Probably a vampire"
-elsif ageCheck != applicantAge && (garlicReply ==false && insuranceReply == false)
-  puts "Almost certainly a vampire"
-elsif fullName == "Drake Cula" || fullName == "Tu Fang"
-  puts "Definitely a vampire"
-else
-  puts "Results inconclusive"
+allergyVamp = nil
+
+  loop do
+    puts "Do you have any allergies? When finished, or if no allergies type done"
+    allergyAnswer = gets.chomp.downcase
+    if allergyAnswer == "sunshine"
+      allergyVamp = true
+      break
+    elsif allergyAnswer == "done"
+      break
+    end
+  end
+
+  currentYear = Time.new.year
+  estimatedAge = currentYear - birthYear
+  finalAge = nil
+  fullName = firstName + " " + lastName
+
+  if estimatedAge.to_i == applicantAge.to_i || (estimatedAge.to_i - 1) ==  applicantAge.to_i
+    finalAge = true
+  else
+    finalAge = false
+  end
+
+  puts "Results for #{[firstName, lastName].compact.join(' ')}:"
+  #Comparison of the Applicant's claimed age versus their estimated age
+  #puts "Estimated Age: #{estimatedAge}"
+  #puts "Claimed Age: #{applicantAge}"
+
+  #Formula checks if the Applicant supplied the correct age, accounting   for if applicant's birthday has not happened.
+  if allergyVamp == true
+    puts "Definitely a vampire"
+  elsif fullName == "Drake Cula" || fullName == "Tu Fang"
+    puts "Definitely a vampire"
+  elsif finalAge == false && (garlicReply == "n" && insuranceReply == " n" )
```

Comment:
There is some refactoring that could be done, but the conditional logic is good. Booleans are much easier to evaluate in conditionals rather than equality to strings. For instance: 
```
if garlic_bread.downcase == "yes" || garlic_bread.downcase == "y"
    garlic_bread = true
else
   garlic_bread = false
end
```
can be refactored to:
```
garlic_bread = garlic_bread.downcase == "yes" || garlic_bread.downcase == "y"
```
If `garlic_bread.downcase` is neither of those answers, `garlic_bread` will be set to false.  Then you can use the boolean variable for `garlic_bread` in your conditionals like:
```
if finalAge == false && (garlic_bread || health_insurance)
```
would be checking to see if the employee lied about their age and either they wanted garlic bread or wanted health insurance. You can also use the `!` operator to check for a false variable like:
```
elsif finalAge! && (garlic_bread! || health_insurance!)
``` 
will check to see if the age was false and either the garlic bread or health insurance was set to false.